state/accumulator.go

```diff
@@ -412,11 +415,40 @@ func (a *Accumulator) Accumulate(txn *sqlx.Tx, userID, roomID string, prevBatch
 			if ev.NID > latestNID {
 				latestNID = ev.NID
 			}
+			// if this is a redaction, try to redact the referenced event (best effort)
+			if !ev.IsState && ev.Type == "m.room.redaction" {
```

Comment:
Are there any rules about who is allowed to issue redactions? I suppose there might be, but they are the upstream homeserver's responsibility to worry about. (The proxy is behaving like a client here.)

Reply:
Yup.